src/main/java/com/code_intelligence/jazzer/mutation/mutator/collection/ChunkCrossOvers.java

```diff
@@ -0,0 +1,248 @@
+      for (int i = 0; i < chunkSize; i++) {
+        Entry<K, V> from = fromIterator.next();
+        Entry<K, V> to = toIterator.next();
+        to.setValue(from.getValue());
+      }
+    });
+  }
+
+  static <K, V> void crossOverChunk(Map<K, V> map, Map<K, V> otherMap,
+      SerializingMutator<K> keyMutator, SerializingMutator<V> valueMutator, PseudoRandom prng) {
+    if (prng.choice()) {
+      crossOverChunkKeys(map, otherMap, keyMutator, prng);
+    } else {
+      crossOverChunkValues(map, otherMap, valueMutator, prng);
+    }
+  }
+
+  private static <K, V> void crossOverChunkKeys(
+      Map<K, V> map, Map<K, V> otherMap, SerializingMutator<K> keyMutator, PseudoRandom prng) {
+    int maxChunkSize = Math.min(map.size(), otherMap.size());
+    withChunk(map, otherMap, maxChunkSize, prng, (fromIterator, toIterator, chunkSize) -> {
+      // Remove keys only after all cross-overs to prevent generating them anew.
+      List<K> keysToRemove = new ArrayList<>(chunkSize);
+      Map<K, V> entriesToAdd = new HashMap<>(chunkSize);
+
+      // If a key cross over results in an already existing key or null, it is
+      // retried up to MAX_FAILED_CROSS_OVER_ATTEMPTS times.
+      for (int i = 0; i < chunkSize; i++) {
+        Entry<K, V> to = toIterator.next();
+        Entry<K, V> from = fromIterator.next();
+
+        // Keys are still in "map" and would be mutated without detaching them
+        // prior to the cross-over. Keys from "otherMap" are not mutated by
+        // the cross-over and so don't need to be detached.
+        K detachedKey = keyMutator.detach(to.getKey());
+        K newKey = attemptCrossOver(detachedKey, from.getKey(), keyMutator, prng,
+            k -> k != null && !map.containsKey(k) && !entriesToAdd.containsKey(k));
+
+        // If the key could be crossed over, replace the whole entry, otherwise
+        // only set the (potentially) new value for the existing key.
+        if (newKey != null) {
+          entriesToAdd.put(newKey, to.getValue());
+          keysToRemove.add(to.getKey());
+        }
+      }
+      keysToRemove.forEach(map::remove);
+      map.putAll(entriesToAdd);
+    });
+  }
+
+  private static <K, V> void crossOverChunkValues(
+      Map<K, V> map, Map<K, V> otherMap, SerializingMutator<V> valueMutator, PseudoRandom prng) {
+    int maxChunkSize = Math.min(map.size(), otherMap.size());
+    withChunkElements(map, otherMap, maxChunkSize, prng, (fromEntry, toEntry, chunkSize) -> {
+      // Values are still in "map" and would be mutated without detaching
+      // them prior to the cross-over. Values from "otherMap" are not mutated
+      // by the cross-over and so don't need to be detached.
+      V detachedValue = valueMutator.detach(toEntry.getValue());
+      // If a cross-over results in the same value, it is retried up to
+      // MAX_FAILED_CROSS_OVER_ATTEMPTS times.
+      V newValue = attemptCrossOver(detachedValue, fromEntry.getValue(), valueMutator, prng,
```

Comment:
I have thought more about whether we should have this retries at all for crossOver. The situation is quite different from `mutate` as `crossOver` is less likely to return new values essentially by design: It combines the two arguments and there is less and less freedom for that the more similar they are, with them being equal being the worst case. 

We could have a short sync on this tomorrow and brainstorm together.

Reply:
As discussed, removed the retry.